test/EtomicSwap.js

```diff
@@ -274,6 +427,83 @@ contract('EtomicSwap', function(accounts) {
+        await this.erc721token.safeTransferFrom(accounts[0], this.swap.address, tokenId, data).should.be.fulfilled;
+
+        // Check the ownership of the token before receiver spend payment - should be owned by swap contract
+        const tokenOwnerBeforeReceiverSpend = await this.erc721token.ownerOf(tokenId);
+        assert.equal(tokenOwnerBeforeReceiverSpend, this.swap.address);
+
+        // Attempt to spend with invalid secret - should fail
+        await this.swap.receiverSpendErc721(id, zeroAddr, this.erc721token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+
+        // Attempt to claim from non-receiver address even with valid secret - should fail
+        await this.swap.receiverSpendErc721(id, secretHex, this.erc721token.address, tokenId, accounts[0], { from: accounts[0] }).should.be.rejectedWith(EVMThrow);
+
+        // Successful spend by receiver with valid secret
+        await this.swap.receiverSpendErc721(id, secretHex, this.erc721token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.fulfilled;
+
+        // Check the state of the payment
+        const payment = await this.swap.payments(id);
+        assert.equal(payment[2].valueOf(), RECEIVER_SPENT);
+
+        // Check the ownership of the token - should be transferred to the receiver (accounts[1])
+        const tokenOwner = await this.erc721token.ownerOf(tokenId);
+        assert.equal(tokenOwner, accounts[1]);
+
+        // Attempting to spend again - should fail
+        await this.swap.receiverSpendErc721(id, secretHex, this.erc721token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
+    });
+
+    it('should allow receiver to spend ERC1155 payment by revealing a secret', async function () {
+        const lockTime = await currentEvmTime() + 1000;
+        const tokenId = 1; // Token ID used in Erc1155Token contract
+        const amountToSend = 2; // Amount of tokens to send
+
+        const data = web3.eth.abi.encodeParameters(
+            ['bytes32', 'address', 'address', 'bytes20', 'uint64'],
+            [id, accounts[1], this.erc1155token.address, secretHash, lockTime]
+        );
+        // Call safeTransferFrom directly to transfer the tokens to the EtomicSwap contract
+        await this.erc1155token.safeTransferFrom(accounts[0], this.swap.address, tokenId, amountToSend, data).should.be.fulfilled;
+
+        // Check the balance of the token before receiver spend payment - should be in swap contract
+        let tokenBalanceBeforeReceiverSpend = await this.erc1155token.balanceOf(this.swap.address, tokenId);
+        assert.equal(tokenBalanceBeforeReceiverSpend.toNumber(), amountToSend);
+
+        // Attempt to spend with invalid secret - should fail
+        await this.swap.receiverSpendErc1155(id, amountToSend, zeroAddr, this.erc1155token.address, tokenId, accounts[0], { from: accounts[1] }).should.be.rejectedWith(EVMThrow);
```

Comment:
`zeroAddr` is 20 bytes instead of 32, please use 32 bytes argument for 100% correctness 🙂

Reply:
added invalidSecretHex